shims/spark32/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -142,7 +145,14 @@ object FileFormatWriter extends Logging {
         Alias(Empty2Null(p), p.name)()
       case attr => attr
     }
-    val empty2NullPlan = if (needConvert) ProjectExec(projectList, plan) else plan
+
+    val empty2NullPlan = if (isVeloxBackend && nativeEnabled) {
+      // Velox backend only support static partition write.
+      // And no need to add sort operator for static partition write.
+      plan
```

Comment:
how do we know this is the static partition write ?

Reply:
@ulysses-you [Here ](https://github.com/oap-project/gluten/pull/3076/files#diff-967086e72dea45b168150635c2f6d9844a7dae173c19dd668bf88b3cbb6872a4R114)we checked only static partition write support in velox backend. So when isVeloxBackend && nativeEnabled enabled, it is the static partition write.